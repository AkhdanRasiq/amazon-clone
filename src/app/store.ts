import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import productReducer from '../features/productSlice'
import alertReducer from '../features/alertSlice'

export const store = configureStore({
  reducer: {
    products  : productReducer,
    alert     : alertReducer
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
