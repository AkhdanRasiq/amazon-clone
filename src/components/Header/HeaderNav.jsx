import ShoppingBasketIcon from "@mui/icons-material/ShoppingBasket"

function HeaderNav() {
  return (
    <div className="headerNav">
      <div className="headerNavOption">
        <span className="headerNavOptionTitle">
          <p>Hello Guest</p>
        </span>
        <span className="headerNavOptionItem">
          <p>Sign in</p>
        </span>
      </div>

      <div className="headerNavOption">
        <span className="headerNavOptionTitle">
          <p>Returns</p>
        </span>
        <span className="headerNavOptionItem">
          <p>& Orders</p>
        </span>
      </div>

      <div className="headerNavOption">
        <span className="headerNavOptionTitle">
          <p>Your</p>
        </span>
        <span className="headerNavOptionItem">
          <p>Prime</p>
        </span>
      </div>

      <div className="headerNavOptionBasket">
        <ShoppingBasketIcon />
        <span className="headerNavOptionItem headerNavBasketCount">
          <p>0</p>
        </span>
      </div>
    </div>
  )
}

export default HeaderNav
