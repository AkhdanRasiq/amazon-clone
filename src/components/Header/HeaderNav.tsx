import React from 'react'

import ShoppingBasketIcon from "@mui/icons-material/ShoppingBasket"
import { Link } from 'react-router-dom'

import { selectProductList } from '../../features/productSlice'
import { useAppSelector } from '../../app/hooks'

function HeaderNav() {
  const productListTotal = useAppSelector(selectProductList).length

  return (
    <div className="headerNav">
      <div className="headerNavOption">
        <span className="headerNavOptionTitle">
          <p>Hello Guest</p>
        </span>
        <span className="headerNavOptionItem">
          <p>Sign in</p>
        </span>
      </div>

      <div className="headerNavOption">
        <span className="headerNavOptionTitle">
          <p>Returns</p>
        </span>
        <span className="headerNavOptionItem">
          <p>& Orders</p>
        </span>
      </div>

      <div className="headerNavOption">
        <span className="headerNavOptionTitle">
          <p>Your</p>
        </span>
        <span className="headerNavOptionItem">
          <p>Prime</p>
        </span>
      </div>

      <Link to="/checkout">
        <div className="headerNavOptionBasket">
          <ShoppingBasketIcon />
          <span className="headerNavOptionItem headerNavBasketCount">
            <p>{productListTotal}</p>
          </span>
        </div>
      </Link>
    </div>
  )
}

export default HeaderNav
