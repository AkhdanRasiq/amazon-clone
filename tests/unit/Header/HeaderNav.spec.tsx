import React from 'react'
import { render } from "@testing-library/react"
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { store } from '../../../src/app/store';
import { Provider } from 'react-redux';

import HeaderNav from "../../../src/components/Header/HeaderNav"

describe("HeaderNav.tsx", () => {
  const history = createMemoryHistory({ initialEntries: ['/'] });

  it("Render Properly", async () => {
    const wrapper = render(
      <Router location={history.location} navigator={history}>
        <Provider store={store}>
          <HeaderNav />
        </Provider>
      </Router>
    )
    await expect(wrapper).toMatchSnapshot()
  })

  it('Routing Checkout', () => {
    const { container } = render(
      <Router location={history.location} navigator={history}>
        <Provider store={store}>
          <HeaderNav />
        </Provider>
      </Router>
    );
    expect(history.location.pathname).toBe('/');

    const btnCheckout: HTMLElement  = container.querySelector('.headerNavOptionBasket') as HTMLElement
    btnCheckout.click()
    expect(history.location.pathname).toBe('/checkout')
  })
})
