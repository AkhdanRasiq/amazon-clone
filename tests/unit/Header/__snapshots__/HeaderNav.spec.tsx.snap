// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNav.tsx Render Properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="headerNav"
      >
        <div
          class="headerNavOption"
        >
          <span
            class="headerNavOptionTitle"
          >
            <p>
              Hello Guest
            </p>
          </span>
          <span
            class="headerNavOptionItem"
          >
            <p>
              Sign in
            </p>
          </span>
        </div>
        <div
          class="headerNavOption"
        >
          <span
            class="headerNavOptionTitle"
          >
            <p>
              Returns
            </p>
          </span>
          <span
            class="headerNavOptionItem"
          >
            <p>
              & Orders
            </p>
          </span>
        </div>
        <div
          class="headerNavOption"
        >
          <span
            class="headerNavOptionTitle"
          >
            <p>
              Your
            </p>
          </span>
          <span
            class="headerNavOptionItem"
          >
            <p>
              Prime
            </p>
          </span>
        </div>
        <a
          href="/checkout"
        >
          <div
            class="headerNavOptionBasket"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ShoppingBasketIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m17.21 9-4.38-6.56c-.19-.28-.51-.42-.83-.42-.32 0-.64.14-.83.43L6.79 9H2c-.55 0-1 .45-1 1 0 .09.01.18.04.27l2.54 9.27c.23.84 1 1.46 1.92 1.46h13c.92 0 1.69-.62 1.93-1.46l2.54-9.27L23 10c0-.55-.45-1-1-1h-4.79zM9 9l3-4.4L15 9H9zm3 8c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
              />
            </svg>
            <span
              class="headerNavOptionItem headerNavBasketCount"
            >
              <p>
                0
              </p>
            </span>
          </div>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="headerNav"
    >
      <div
        class="headerNavOption"
      >
        <span
          class="headerNavOptionTitle"
        >
          <p>
            Hello Guest
          </p>
        </span>
        <span
          class="headerNavOptionItem"
        >
          <p>
            Sign in
          </p>
        </span>
      </div>
      <div
        class="headerNavOption"
      >
        <span
          class="headerNavOptionTitle"
        >
          <p>
            Returns
          </p>
        </span>
        <span
          class="headerNavOptionItem"
        >
          <p>
            & Orders
          </p>
        </span>
      </div>
      <div
        class="headerNavOption"
      >
        <span
          class="headerNavOptionTitle"
        >
          <p>
            Your
          </p>
        </span>
        <span
          class="headerNavOptionItem"
        >
          <p>
            Prime
          </p>
        </span>
      </div>
      <a
        href="/checkout"
      >
        <div
          class="headerNavOptionBasket"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ShoppingBasketIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m17.21 9-4.38-6.56c-.19-.28-.51-.42-.83-.42-.32 0-.64.14-.83.43L6.79 9H2c-.55 0-1 .45-1 1 0 .09.01.18.04.27l2.54 9.27c.23.84 1 1.46 1.92 1.46h13c.92 0 1.69-.62 1.93-1.46l2.54-9.27L23 10c0-.55-.45-1-1-1h-4.79zM9 9l3-4.4L15 9H9zm3 8c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
            />
          </svg>
          <span
            class="headerNavOptionItem headerNavBasketCount"
          >
            <p>
              0
            </p>
          </span>
        </div>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
